src/ripple/shamap/impl/SHAMapDelta.cpp

```diff
@@ -143,11 +143,11 @@ bool SHAMap::compare (SHAMap::ref otherMap, Delta& differences, int maxCount)
         if (ourNode->isLeaf () && otherNode->isLeaf ())
         {
             // two leaves
-            if (ourNode->getTag () == otherNode->getTag ())
+            if (ourNode->peekItem()->getTag () == otherNode->peekItem()->getTag ())
```

Comment:
I never understood this `peek` thing in rippled.  It took me quite a while that it just meant "get".

The idea of "peeking" comes from tokenizers - which spit out one token at a time.  If you want to look at the _next_ token, the one that hasn't arrived yet, you "peek" at this token.

That makes sense - but the way we use it in ripple it seems to only mean "get" - so why don't we just use "get"?

Also, "item" is meaningless.  I look at this code and I have no idea what sort of datatype is there.  I go to look at the accessor, and it's returning `mItem`, which is of type `SHAMapItem`, which is at least consistent, but really tells me nothing.  It's beyond the scope of this change, but it'd be great if we had a name that said what was going on.